walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,90 @@
+      for(Event e : events){ //looks at every event
+          Collection<String> eventAtt = e.getAttendees();
+          for(String a : eventAtt){ //looks at all the people attending the event
+                System.out.println(a + eventAtt.size() + " " + attendees.contains(a));
+              if(attendees.contains(a)){
+                  System.out.println("Goes in");
+                  busyTimes.add(e.getWhen());
+                  break; //found someone in this event, we can stop searching this one
+              }
+          }
+
+      }
+      List<TimeRange> tr = new ArrayList(); 
+
+      //Assume whole day is free at first
+    tr.add(TimeRange.WHOLE_DAY);
+    Collections.sort(busyTimes, TimeRange.ORDER_BY_START); //sort the busy times by start date for easier parsing
+        for(TimeRange e : busyTimes){
+            int origLen = tr.size();
+            for(int i = 0; i < origLen; i ++){
+                TimeRange freeTime = tr.get(i);
+                if(tr.get(i).contains(e)){ //remove fully, split free by e
+                    TimeRange secHalf = TimeRange.fromStartDuration(e.end(), freeTime.end() - e.end());
+                    TimeRange firstHalf = TimeRange.fromStartDuration(freeTime.start(), freeTime.duration() - secHalf.duration() - e.duration());
+                    tr.remove(freeTime); //remove old time
+                    i--; //back the counter up because we removed an item
+                    origLen --; //we need to view one less item
+                    if(firstHalf.duration() >= dur) //only add to list if the free time >= duration wanted
+                        tr.add(firstHalf);
+                    if(secHalf.duration() >= dur)
+                        tr.add(secHalf);
+                } else if(freeTime.overlaps(e)){ //starts are the same, but ends arent, cut free by some amount
+                    int newStart = 0;
+                    int newEnd = 0;
+                    //There are two cases, if the free time occurs before or after the busy time
+                    if(freeTime.start() < e.start()){
+                        newStart = freeTime.start();
+                        newEnd = freeTime.duration() - (freeTime.end() - e.start());//freeTime.end() - (freeTime.end() - e.start());
+                    } else {
+                        newStart = e.end();
+                        newEnd = freeTime.duration() - (e.end() - freeTime.start());
+                    }
+                    TimeRange cutTime = TimeRange.fromStartDuration(newStart, newEnd);
+                    tr.remove(freeTime);
+                    i--; //back the counter up because we removed an item
+                    origLen --; //we need to view one less item
+                    if(cutTime.duration() >= dur)
+                    tr.add(cutTime);
+                }
+            }
+        }
+    Collections.sort(tr, TimeRange.ORDER_BY_START); //Order by start times
+      Collection<TimeRange> collTr = tr; //Cast to a Collection
```

Comment:
hmm - do you need this cast to a Collection? List should be a subinterface of Collection: https://docs.oracle.com/javase/8/docs/api/java/util/List.html